### DISCUSSION TITLE
quarto revealjs presentation: automatic outline and section break

### DISCUSSION THREAD

**Discussion by shannonpileggi** [quarto-dev/quarto-cli#1492] [Q&A]
Hi! One thing I liked from my old latex/beamer presentations were slides that automatically created a numbered (or not numbered) presentation outline based on new sections, and then each time you entered a new section you would see the full outline with the other sections grayed out and the current section highlighted. Is there a way to mimic this automation in quarto revealjs presentations?

**Comment by mcanouil** [comment]
I don't know if there is a simple way to do that in Quarto, but I am aware of a reveal.js plugin that does (almost) what you want by adding a footer: https://github.com/e-gor/Reveal.js-TOC-Progress
The idea would be to modify the layout and add the content of the footer as a slide before any level 1 or 2 headers.

**Reply by shannonpileggi** [reply]
Thank you for sharing that option! That is neat, but takes up a bit of real estate on the screen.

**Reply by mcanouil** [reply]
I've lust looked at @rweekly and there is one post by @meghall06 (i.e., <https://meghan.rbind.io/blog/quarto-slides/#section-header>) where she is using a header outline.
Although, this is a pretty manual modification: https://github.com/meghall06/personal-website/blob/a5f23e71bdfcee09a5d852befe9a6a6d6a4aff01/static/slides/NEAIR/NEAIR.qmd#L45-L49

PS: I am interested in this question, so when I have time, I'll probably work (unless someone else take this on him/herself) on an extension to add such outline after "part" section for example.

**Reply by shannonpileggi** [reply]
Thanks, that is also a great way of doing it! I look forward to seeing what you or others come up with!

**Reply by mcanouil** [reply]
I just found out about an extension for Quarto which seems to be a good fit: https://github.com/andrie/reveal-auto-agenda

**Reply by andrie** [reply]
It's a good fit because I wrote it in response to this thread :-)
Please let me know how well it works for you.

**Comment by andrie** [comment]
I've created an extension that does this at https://github.com/andrie/reveal-auto-agenda
